### PR TITLE
Create NNNN-atf_facade_proposal.md

### PR DESCRIPTION
**Pull request by aderiabin**
This proposal describes unified interface for ATF (Automated Test Framework) - ATF Facade.
ATF should provide unified and flexible interface for creation test scripts to perform checks of SDL functionality.

Main ATF Facade purposes:
 - Simplify access to ATF features for test scripts
 - Separate test scripts from ATF
 - Provide well-structured intuitive interface of ATF for test scripts
 - Unify way of use ATF for creating automated tests
 - Create single point of ATF features documentation
 - Provide ability to create more human-readable tests

### CODE REVIEW

File: proposals/NNNN-atf_facade_proposal.md [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+ - To separate test scripts from ATF
+ - Provide well-structured intuitive interface of ATF for test scripts
+ - Unify way of use ATF for creating automated tests
+ - Create single point of ATF features documentation
+ - To provide ability to create more human-readable tests
+
+## Motivation
+
+Currently ATF has a lot of functionality for testing most of SDL features, but each automated test uses different ATF modules.
+ATF does not have unified interface for provide its functionality for test scripts. Instead that ATF has many separated modules with its own interfaces which is used for create automated scripts for SDL testing purposes.
```
Comment (dboltovskyi) [inline proposals/NNNN-atf_facade_proposal.md]: @aderiabin Suggest to change 'interface for provide' to 'interface to provide'

Comment (dboltovskyi) [reply proposals/NNNN-atf_facade_proposal.md]: @aderiabin Suggest to change 'which is used for create automated scripts ...' to 'which is used in automated scripts.'

File: proposals/NNNN-atf_facade_proposal.md [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+So current approach leads to create test scripts which have multiple direct dependencies on different modules of ATF.
+In other words automated test scripts are fully dependent from ATF implementation and each change in ATF can lead to need for change of test scripts.
+
+In addition current approach demand a strong knowledge in ATF modules purposes and interactions between them for create good automated test scenario using ATF.
```
Comment (dboltovskyi) [inline proposals/NNNN-atf_facade_proposal.md]: @aderiabin Suggest to change 'demand' to 'require'

File: proposals/NNNN-atf_facade_proposal.md [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+So creation of automated test scripts take a lot of time and it can be a bit complicated for Mobile application developers in the beginning.
+
+Creation of ATF Facade will allow:
+ - to create test scripts which are independent from ATF implementation
+ - to unify ATF use approach for test script creation
+ - to document ATF functionality create test script creation
+ - to improve tests script readability and its structure on base of scenarious
```
Comment (dboltovskyi) [inline proposals/NNNN-atf_facade_proposal.md]: @aderiabin Typo 'scenarious'

**Review by dboltovskyi — APPROVED** [review]

File: proposals/NNNN-atf_facade_proposal.md (hunk 1 above)
Comment (aderiabin) [reply proposals/NNNN-atf_facade_proposal.md]: Done.

File: proposals/NNNN-atf_facade_proposal.md (hunk 2 above)
Comment (aderiabin) [reply proposals/NNNN-atf_facade_proposal.md]: Done.

**Review by AByzhynar — APPROVED** [review]

**Review by GetmanetsIrina — APPROVED** [review]

**Review by AKalinich-Luxoft — APPROVED** [review]

File: proposals/NNNN-atf_facade_proposal.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+## Proposed solution
+
+Add new module Facade with next submodules in ATF:
+ - Runner - Manage of run test scenario
+ - SDL - Provide access to manage SDL
+ - HMI - Provide ability to emulate HMI and manage its expectations
+ - Mobile - Provide ability to emulate Mobile device and manage its expectations
+ - Utilities - Provide access to utilities of ATF
+ - Constants - Provide access to all constants used in ATF
```
Comment (LuxoftAKutsan) [inline proposals/NNNN-atf_facade_proposal.md:46]: @aderiabin what about config? Expectations?

File: proposals/NNNN-atf_facade_proposal.md [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+### Submodules description
+
+#### Runner - Manage of run test scenario
+
+ - Step - Add step to test scenario
+ - Title - Add titled section to test scenario
+ - TestSettings - Set special test settings for run of current test
+
+#### SDL - Provide access to manage SDL
+
```
Comment (LuxoftAKutsan) [inline proposals/NNNN-atf_facade_proposal.md:57]: @aderiabin please expand description. What parameters should be passed? How SDL will be stoped\started ...

File: proposals/NNNN-atf_facade_proposal.md [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+ - StartSDL - Start SDL
+ - StopSDL - Stop SDL
+ - CheckStatusSDL - Check current status of SDL
```
Comment (LuxoftAKutsan) [inline proposals/NNNN-atf_facade_proposal.md:60]: @aderiabin  What is the result of this function ?

File: proposals/NNNN-atf_facade_proposal.md [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+
+ - Connect - Create emulated Mobile device and connect it to SDL
+ - Disconnect - Disconnect Mobile device from SDL
+ - StartSession - Start mobile session on Mobile device
+ - StartService - Start service on mobile session
+ - StopService - Stop service on mobile session
+ - Send - Send raw message from mobile session
+ - SendRPC - Send RPC message from mobile session
+ - ExpectResponse - Create expectation for response from SDL
+ - ExpectNotification - Create expectation for notification from SDL
```
Comment (LuxoftAKutsan) [inline proposals/NNNN-atf_facade_proposal.md:84]: @aderiabin Expect packet?

File: proposals/NNNN-atf_facade_proposal.md [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+
+#### Utilities - Provide access to utilities of ATF
+
```
Comment (LuxoftAKutsan) [inline proposals/NNNN-atf_facade_proposal.md:87]: @aderiabin not clear what exact functionality should be provided

**Review by LuxoftAKutsan — CHANGES_REQUESTED** [review]
Please describe interfaces more. Parameters and results are not clear

**Comment by LuxoftAKutsan** [comment]
Also please provide description of PR

aderiabin:
Description was provided.

File: proposals/NNNN-atf_facade_proposal.md (hunk 4 above)
Comment (aderiabin) [reply proposals/NNNN-atf_facade_proposal.md:46]: @LuxoftAKutsan Will not be added to base Facade. Can be added in addition.

File: proposals/NNNN-atf_facade_proposal.md (hunk 5 above)
Comment (aderiabin) [reply proposals/NNNN-atf_facade_proposal.md:57]: @LuxoftAKutsan Will not be added to base Facade. Can be added in addition.

File: proposals/NNNN-atf_facade_proposal.md (hunk 6 above)
Comment (aderiabin) [reply proposals/NNNN-atf_facade_proposal.md:60]: @LuxoftAKutsan Will not be added to base Facade. Can be added in addition.

File: proposals/NNNN-atf_facade_proposal.md (hunk 7 above)
Comment (aderiabin) [reply proposals/NNNN-atf_facade_proposal.md:84]: @LuxoftAKutsan Will not be added to base Facade. Can be added in addition.

File: proposals/NNNN-atf_facade_proposal.md (hunk 8 above)
Comment (aderiabin) [reply proposals/NNNN-atf_facade_proposal.md:87]: @LuxoftAKutsan Will not be added to base Facade. Can be added in addition.

**Review by LuxoftAKutsan — APPROVED** [review]

**Pull request merged**